hbase-common/src/main/java/org/apache/hadoop/hbase/nio/RefCnt.java

```diff
@@ -35,7 +35,7 @@
 @InterfaceAudience.Private
 public class RefCnt extends AbstractReferenceCounted {
 
-  private static final ResourceLeakDetector<RefCnt> detector =
+  public static final ResourceLeakDetector<RefCnt> detector =
```

Comment:
nit: Do we really need this public?
I saw that you used it in unit test. If that is the reason, can't we use reflection to access the private field?

Reply:
I don't think it's reasonable to use reflection in the tests for this. We will need to access it in several tests, and maybe in other parts of the code too, and the whole class is marked as IA private, so I would rather keep it simple and set this public